discord_bots/cogs/economy.py

```diff
@@ -0,0 +1,1119 @@
+                            None,
+                        )
+
+                        try:
+                            await EconomyCommands.create_transaction(
+                                winning_prediction.in_progress_game,
+                                winning_prediction.player,
+                                win_value,
+                                winning_prediction,
+                            )
+                        except Exception as e:
+                            embed.insert_field_at(
+                                index=0,
+                                name="",
+                                value=f"Prediction resolution failed for <@{winning_prediction.player_id}> | Win Value = {win_value} | Exception: {e}",
+                                inline=False
+                            )    
+                            pass
+                        else:
+                            player: Player = (
+                                session.query(Player)
+                                .filter(Player.id == winning_prediction.player_id)
+                                .first()
+                            )
+                            player.currency += win_value
+                            winning_prediction.outcome = True
+                            session.commit()
+
+                            # Adds win_value to player in dictionary, or creates new dict item
+                            # Combines multiple predictions into one win value to be returned
+                            # Mutliple transactions are still created (one per prediction)
+                            if any(
+                                x == winning_prediction.player_id
+                                for x in iter(summed_winners.keys())
+                            ):
+                                summed_winners[winning_prediction.player_id] += win_value
+                            else:
+                                summed_winners[winning_prediction.player_id] = win_value
+
+                    sorted_winners: dict = dict(reversed(sorted(summed_winners.items(), key=itemgetter(1))))
+                    sorted_losers: dict = dict(reversed(sorted(summed_losers.items(), key=itemgetter(1))))
+                    
+                    prediction_winners: str = ">>> "
+                    prediction_losers: str = ">>> "
+                    for key, value in list(sorted_winners.items())[:10]:
+                        prediction_winners += f"<@{key}> **+{round(value, None)}** {CURRENCY_NAME}\n"
+                    for key, value in list(sorted_losers.items())[:10]:
+                        prediction_losers += f"<@{key}> **-{round(value, None)}** {CURRENCY_NAME}\n"
+
+                    if len(sorted_winners) > 10:
+                        prediction_winners += "..."
+                    if len(sorted_losers) > 10:
+                        prediction_losers += "..."
```

Comment:
this is a really nice touch. love the little touches you've added to everything

Reply:
Thanks!